### PR TITLE
[MISC] Move ostream iterator out of std/ranges header.

### LINKED ISSUE #988: We have std::ranges::ostream_iterator etc imported into std/ranges. But I couldn't find any reference that they will be added with c++20.
In this case we need to remove them and implement them in SeqAn.

Port the ostream iterator to seqan3/io/stream. Alias the std::ostream if ranges feature test macro is set otherwise forward to ranges::ostream_iterator.

### PR DESCRIPTION
**Pull request by rrahn**
Fixes #988

### CODE REVIEW

File: include/seqan3/io/stream/iterator.hpp [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+/*!\file
+ * \brief Provides seqan3::ostream and seqan3::ostreambuf iterator.
+ * \author Rene Rahn <rene.rahn AT fu-berlin.de>
+ */
+
+#pragma once
+
+#ifdef __cpp_lib_ranges
+#include <iterator>
```
Comment (h-2) [inline include/seqan3/io/stream/iterator.hpp]: you need to include the header before checking the macro, otherwise it may not be defined.

Comment (rrahn) [reply include/seqan3/io/stream/iterator.hpp]: Ok, I included <version> which will provide all feature test macros.

Comment (h-2) [reply include/seqan3/io/stream/iterator.hpp]: version is c++20, it broke your builds... You can just include iterator undconditionally...

Comment (rrahn) [reply include/seqan3/io/stream/iterator.hpp]: Yeah, I just realized that. Haven’t read the notes. Could have been tagged for c++20 more clearly on cppreference though 😒

**Comment by rrahn** [comment]
travis linker error on filsystem. otherwise good to go

**Pull request merged**